### DISCUSSION TITLE
basic_voice.py

### DISCUSSION THREAD

**Discussion by christos-f** [Rapptz/discord.py#5912] [General]
async def from_url(cls, url, *, loop=None, stream=False):
        loop = loop or asyncio.get_event_loop()
        data = await loop.run_in_executor(None, lambda: ytdl.extract_info(url, download=not stream))

        if 'entries' in data:
            # take first item from a playlist
            data = data['entries'][0]

         # ***Some youtube videos will cause data['url'] to produce a manifest url as seen below***

        if len(str(data['url']).split('https://manifest.googlevideo.com')) > 1:

            # ***This seems to fix when above issue occurs***
            filename = data['fragment_base_url']

        else:
            filename = data['url']
             
            # ***Took this out because i only want to stream, never download***
        #filename = data['url'] if stream else ytdl.prepare_filename(data)
        return cls(discord.FFmpegPCMAudio(filename, **ffmpeg_options), data=data)

Some youtube videos my friends have tried caused data['url'] to produce url that will not play through the discord bot.
I have not tried other websites, so i believe this may be a youtube specific problem.
I am wondering if this has occured for anyone else, and if there could be a better way to handle the issue.

**Comment by Fisch37** [comment]
Could you provide an example of such a video?
